### PR TITLE
Machines docs: API docs update, nav changes, split content into new files

### PR DESCRIPTION
**Pull request by andie787**
### Summary of changes

- split API docs into separate pages
- move Machines states into a new page (because Machine states are not just an API thing)
- add new nav sections for API docs and Reference
- add volumes API docs draft

### Related Fly.io community and GitHub links
n/a

### Notes
n/a

### CODE REVIEW

File: machines/api-machines-resource.html.markerb [hunk 1]
```diff
@@ -0,0 +1,463 @@
...
+
+## List all Machines for an app
+
+`GET /apps/{app_name}/machines`
+
+### Query parameters for list Machines
+
+Append parameters to filter the response.
+
+**`state`**: Filter by the state of the Machines in the app. For example, to return only Machines in the `started` state: `GET /apps/my-app-name/machines?started`
```
Comment (benwaffle) [inline machines/api-machines-resource.html.markerb:35]: This parameter doesn't exist / work

File: partials/_machines_nav.html.erb [hunk 2]
```diff
@@ -7,16 +7,41 @@
...
+      Control Machines with flyctl
+    </span>
+  <ul>  
+   <li>
     <%= nav_link "Run a new Machine", "/docs/machines/run/" %>
-  </li>
+   </li>
+   <li>
+    <%= nav_link "Update a Machine", "/docs/machines/run/" %>
+   </li>
```
Comment (catflydotio) [inline partials/_machines_nav.html.erb:19]:
```suggestion
```

I'm actively drafting a m update doc but there isn't one to link to yet, right?

Comment (andie787) [reply partials/_machines_nav.html.erb]: no! I removed this for now, but I want to keep the section

File: machines/api-volumes-resource.html.markerb [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+| created_at | string | The date and time the volume was created. |
+| encrypted | boolean | Whether the volume is encrypted. Default true. |
+| fstype | string | n/a |
+| id | string | The volume ID. |
+| name | string | The volume name. |
+| region | string | The region where the volume resides, or the target region for volume create. |
+| size_gb | int | The size of the volume in GB. |
+| snapshot_retention | int | Not yet configurable. The number of days to retain daily snapshots. |
+| state | string | The state of the volume. |
+| zone | string | The hardware zone on which the volume resides. |
```
Comment (catflydotio) [inline machines/api-volumes-resource.html.markerb:28]:
```suggestion
| `attached_alloc_id` | string | n/a |
| `attached_machine_id` | string | The ID of the Machine that’s attached to the volume. |
| `block_size` | int |  |
| `blocks` | int |  |
| `blocks _avail` | int |  |
| `blocks_free` | int |  |
| `created_at` | string | The date and time the volume was created. |
| `encrypted` | boolean | Whether the volume is encrypted. Default true. |
| `fstype` | string | n/a |
| `id` | string | The volume ID. |
| `name` | string | The volume name. |
| `region` | string | The region where the volume resides, or the target region for volume create. |
| `size_gb` | int | The size of the volume in GB. |
| `snapshot_retention` | int | Not yet configurable. The number of days to retain daily snapshots. |
| `state` | string | The state of the volume. |
| `zone` | string | The hardware zone on which the volume resides. |
```

**Pull request merged**